flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,204 @@
+
+plugin_name = 'mqtt'
+logger = logger.bind(name=plugin_name)
+
+class MQTTNotifier:
+    """
+    Example::
+      notify:
+        entries:
+          via:
+            - mqtt:
+                broker_address: "iot.eclipse.org"
+                topic: "flexget/notifications"
+                [broker_port: 1883]
+                [broker_timeout: 30]
+                [broker_transport: ['tcp','websockets'] ]
+                [broker_protocol: ['MQTTv31', 'MQTTv311'] ]
+                [username: yourUsernameHere]
+                [password: yourPasswordHere]
+                [enable_encrypted_communication: True/False]
+                [certificates:
+                    broker_ca_cert: /path/to/pem/encoded/broker_ca_certificate.crt
+                    client_cert: /path/to/pem/encoded/client_certificate.crt
+                    client_key: /path/to/pem/encoded/client_certificate.key
+                    validate_broker_cert: True/False
+                    tls_version: ['tlsv1.2', 'tlsv1.1', 'tlsv1', '']
+                ]
+                [qos: [0,1,2] ]
+                [retain: True/False]
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'topic': {'type': 'string'},
+            'broker_address': {'type': 'string'},
+            'broker_port': {'type': 'integer', 'default': 1883},
+            'broker_timeout': {'type': 'integer', 'default': 30},
+            'broker_transport': {'type': 'string', 'default': 'tcp', 'enum': ['tcp', 'websockets']},
+            'broker_protocol': {'type': 'string', 'default': 'MQTTv311', 'enum': ['MQTTv31', 'MQTTv311']},
+            'username': {'type': 'string', 'default': ''},
+            'password': {'type': 'string', 'default': ''},
+            'enable_encrypted_communication': {'type': 'boolean', 'default': False},
+            'certificates': {
+                'type': 'object',
+                'properties': {
+                    'broker_ca_cert': {'type': 'string', 'default': ''},
+                    'client_cert': {'type': 'string', 'default': ''},
+                    'client_key': {'type': 'string', 'default': ''},
+                    'validate_broker_cert': {'type': 'boolean', 'default': True},
+                    'tls_version': {'type': 'string', 'default': '', 'enum': ['tlsv1.2', 'tlsv1.1', 'tlsv1', '']},
+                },
+                'additionalProperties': False,
+            },
+            'qos': {'type': 'integer', 'minimum': 0, 'maximum': 2, 'default': 0},
+            "retain": {"type": "boolean", 'default': False},
+        },
+        'additionalProperties': False,
+        'required': ['broker_address','topic'],
+    }
+
+    def notify(self, title, message, config):
+        """
+        Publish to an MQTT topic
+        """
+        try:
+            import paho.mqtt.client as mqtt
+            from hashlib import sha256
+            import ssl
+            import time
+
+        except ImportError as e:
```

Comment:
please wrap only the 3rd party dep here and move the stdlib ones to the module level

Reply:
Done with commit 92edfa8285ccd62b3389bb889345c87be42a3cdc